prov/efa/src/rxr/rxr.h

```diff
@@ -262,13 +262,11 @@ enum rxr_rx_comm_type {
 	RXR_RX_WAIT_ATOMRSP_SENT, /* rx_entry wait for atomrsp packet sent completion */
 };
 
-/* peer is in backoff, not allowed to send */
-#define RXR_PEER_IN_BACKOFF (1ULL << 0)
-/* peer backoff was increased during this loop of the progress engine */
-#define RXR_PEER_BACKED_OFF (1ULL << 1)
-#define RXR_PEER_REQ_SENT BIT_ULL(0)
+#define RXR_PEER_REQ_SENT BIT_ULL(0) /* sent a REQ to the peer, peer should send a handshake back */
 #define RXR_PEER_HANDSHAKE_SENT BIT_ULL(1)
 #define RXR_PEER_HANDSHAKE_RECEIVED BIT_ULL(2)
+#define RXR_PEER_IN_BACKOFF BIT_ULL(3) /* peer is in backoff, not allowed to send */
```

Comment:
If we are truly merging the rnr states with the peer state, should we not extend `rxr_peer_state` enum?

Reply:
enum won't work here because these flags are not mutually exclusive. For example, the handshake_sent status does not coincide with handshake received status. If we use enum, that would require 4 enum:
 ```handshake_not_sent_not_received,
handshake_sent_not_received,
handshake_not_sent_received,
handshake_sent_received```)

Reply:
Feeling pedantic this afternoon (effects of _social distancing_) :) A finite-state machine should always be in exactly one state of all possible states, and by definition has to be mutually exclusive. Flags are fair game for bitmaps though. If those are the four combinations of bits they can take on at any transition, we should do what you suggest and make it an enum.

Reply:
I think the variable name ```state``` is misleading here, we are not using a state machine here.  Change it to ```flags``` would be more appropriate ?

Reply:
I think changing the name to `flags` is a good idea.

Reply:
Sure, will do.